j0i/PDB-Home-Progress-Viewer
# chat functionality

# Overview
Added the chat functionality, complete with uploading images from the device's photo reel. Unfortunately there doesn't seem to be a way to make clickable hyperlinks in SwiftUI for iOS 13, so we may have to examine alternatives or give up this feature

 # Testing:
 1. Reach the chat page on any project (they are all hooked up to a hardcoded project at the moment)
 2. Scroll through the list of messages already in the app. Some of them (the ones from you) should appear on the right of the screen in blue. Others will appear on the left in white. Messages from others will have their name above it, unless there are consecutive messages from the same sender.
 3. Verify that the images on some of the messages load
 4. Send a message of your own by typing it into the text box and hitting enter
 5. Verify that the message was added to the server, either by looking at the Firebase console or seeing it update on your device. Note that the simulator doesn't always work for this. Backing out of the project and reselecting it seems to refresh the message list, though.
 6. Add a picture to your message. Tap the photo icon, then select an image.
 7. Remove the image from your message by double tapping it
 8. Add the image back, enter some text, and hit enter.
 9. Verify that your message and the image were uploaded to the server in the same way as in step 5. You can also view the image in the Storage section of the Firebase console, under messageData/\<projectID\>/
